kymatio/scattering3d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,160 @@
+
+        $\\sqrt{\\sum_m (x \\star \\psi_{j,l,m})^2)}$
+
+        if set to False the first order moduli take the form:
+
+        $x \\star \\psi_{j,l,m}$
+
+        The second order moduli change analogously
+        Defaut: True
+    method: string, optional
+        specifies the method for obtaining scattering coefficients
+        ("standard","local","integral"). Default: "standard"
+    points: array-like, optional
+        List of locations in which to sample wavelet moduli. Used when
+        method == 'local'
+    integral_powers: array-like
+        List of exponents to the power of which moduli are raised before
+        integration. Used with method == 'standard', method == 'integral'
+
+    """
+    def __init__(self, J, shape, L=3, sigma_0=1, max_order=2, rotation_covariant=True, method='standard', points=None,
+                 integral_powers=(0.5, 1., 2.), backend=None):
+        super(HarmonicScattering3DTorch, self).__init__()
+        self.J = J
+        self.shape = shape
+        self.L = L
+        self.sigma_0 = sigma_0
+
+        self.max_order = max_order
+        self.rotation_covariant = rotation_covariant
+        self.method = method
+        self.points = points
+        self.integral_powers = integral_powers
+        self.backend = backend
+
+        self.build()
+
+    def build(self):
+        self.M, self.N, self.O = self.shape
+        self.filters = solid_harmonic_filter_bank(
+                            self.M, self.N, self.O, self.J, self.L, self.sigma_0)
+        self.gaussian_filters = gaussian_filter_bank(
+                                self.M, self.N, self.O, self.J + 1, self.sigma_0)
+
+        # transfer the filters from numpy to torch
+        for k in range(len(self.filters)):
+            self.filters[k] = torch.from_numpy(self.filters[k]).type(torch.Tensor)
+            self.register_buffer('tensor' + str(k), self.filters[k])
+        self.gaussian_filters = torch.from_numpy(self.gaussian_filters).type(torch.Tensor)
+        self.register_buffer('tensor_gaussian_filter', self.gaussian_filters)
+
+        methods = ['standard', 'local', 'integral']
+        if (not self.method in methods):
+            raise (ValueError('method must be in {}'.format(methods)))
+        if self.method == 'integral':\
+            self.averaging =lambda x,j: self.backend.compute_integrals(self.backend.fft(x, inverse=True)[...,0],
+                                                                       self.integral_powers)
+        elif self.method == 'local':
+            self.averaging = lambda x,j:\
+                self.backend._compute_local_scattering_coefs(x,
+                        self.tensor_gaussian_filter[j+1], self.points)
+        elif self.method == 'standard':
+            self.averaging = lambda x, j:\
+                self.backend._compute_standard_scattering_coefs(x,
+                        self.tensor_gaussian_filter[j], self.J, self.backend.subsample)
```

Comment:
I'm a bit suspicious about this line. Shouldn't it be `self.tensor_gaussian_filter[J]` or a similar instruction? Similarly, I find it is too bad to pass self.J in this one and not in the previous function. I think it is better to pass self.tensor_gaussian_filter and then to select the j-th filter.

Reply:
@MuawizChaudhary

Reply:
It should indeed be the J'th filter and not the j'th filter, if we are keeping in line with the current version of the software: https://github.com/kymatio/kymatio/blob/master/kymatio/scattering3d/scattering3d.py#L163.

I believe that the previous function we want to pass in the lower case j, and not capital J, see: https://github.com/kymatio/kymatio/blob/master/kymatio/scattering3d/scattering3d.py#L180. .
I'll get on correcting the code and making the functions get the appropriate filter passed in.